gnomad_qc/v4/assessment/calculate_per_sample_stats.py

```diff
@@ -429,28 +733,35 @@ def compute_agg_sample_stats(
         subset += [subset_expr] if by_subset else []
         gen_anc += [meta_s.population_inference.pop] if by_ancestry else []
 
-    ht = ht.transmute(
-        subset=subset,
-        gen_anc=gen_anc,
-        summary_stats=hl.zip(ht.summary_stats_meta, ht.summary_stats),
-    )
+    ht = ht.annotate(subset=subset, gen_anc=gen_anc)
+    ht = ht.explode("gen_anc").explode("subset")
 
-    ht = ht.explode("summary_stats").explode("gen_anc").explode("subset")
-
-    ht = ht.group_by("subset", "gen_anc", variant_filter=ht.summary_stats[0]).aggregate(
-        **{
-            metric: hl.struct(
-                mean=hl.agg.mean(ht.summary_stats[1][metric]),
-                min=hl.agg.min(ht.summary_stats[1][metric]),
-                max=hl.agg.max(ht.summary_stats[1][metric]),
-                quantiles=hl.agg.approx_quantiles(
-                    ht.summary_stats[1][metric], [0.0, 0.25, 0.5, 0.75, 1.0]
-                ),
-            )
-            for metric in ht.summary_stats[1]
-            if isinstance(ht.summary_stats[1][metric], hl.expr.NumericExpression)
-        }
+    ht = ht.group_by("subset", "gen_anc").aggregate(
+        summary_stats=hl.agg.array_agg(
+            lambda x: hl.struct(
+                **{
+                    m: hl.struct(
+                        mean=hl.agg.mean(x[m]),
+                        min=hl.agg.min(x[m]),
+                        max=hl.agg.max(x[m]),
+                        quantiles=hl.agg.approx_quantiles(
```

Comment:
Do we care to add logic for is_nan here?

Reply:
sure, can you add as a suggestion if you already have in mind what works?